bundle/edu.gemini.gsa.client/src/main/scala/edu/gemini/gsa/client/api/GsaParams.scala

```diff
@@ -56,23 +65,14 @@ object GsaParams {
   def get(target: Target, blueprint: BlueprintBase): Option[GsaParams] =
     blueprint match {
       case g: GeminiBlueprintBase => getGeminiParams(target, g)
-      case _                      => None
+      case _                      => none
     }
 
   private def getGeminiParams(target: Target, blueprint: GeminiBlueprintBase): Option[GsaParams] =
     target match {
-      case s: SiderealTarget    => Some(GsaSiderealParams(s.coords, GSAInstrument(blueprint.instrument)))
-      case n: NonSiderealTarget => Some(GsaNonSiderealParams(n.name, GSAInstrument(blueprint.instrument)))
-      case _                    => None
+      case s: SiderealTarget    => GSAInstrument(blueprint.instrument).flatMap(k => GsaSiderealParams(s.coords, k).some).orElse(GsaUnsupportedParams.some)
```

Comment:
Maybe I'm misunderstanding something here, but couldn't this just be:
`GSAInstrument(blueprint.instrument).fold(GsaUnsupportedParams)(GsaSiderealParams(s.coords, _)).some` to make it easier and not need the `flatMap`?